pkg/test/server.go

```diff
@@ -43,9 +42,9 @@ const (
 type Hasher struct {
 }
 
-// Hash function that always returns same value.
-func (h Hasher) Hash(*core.Node) (cache.Key, error) {
-	return cache.Key(node), nil
+// ID function that always returns the same value.
+func (h Hasher) ID(*core.Node) string {
+	return node
```

Comment:
Also need a callback to indicate that a new Envoy has connected to the server or an envoy has disconnected from the server. And a callback that indicates whether the snapshot has been fully transmitted and acked by envoy.

Reply:
There is a callback for various way Envoy connects to the server. There's also a callback for stream disconnect and fetch response. Snapshot transmission callback is WIP.